hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -818,26 +835,28 @@ protected boolean deleteRecord(Data key, int completionId, String source) {
     protected boolean deleteRecord(Data key, int completionId, String source, String origin) {
```

Comment:
Can't we re-write this method like this? Is `onDeleteRecordError` really useful? Seems without it we should be ok. 

```
protected boolean deleteRecord(Data key, int completionId, String source, String origin) {
        R removedRecord = null;
        try {
            removedRecord = doRemoveRecord(key, source);

            if (isEventsEnabled()) {
                Data eventDataKey = toEventData(key);
                Data eventDataValue = toEventData(removedRecord);

                publishEvent(createCacheRemovedEvent(eventDataKey, eventDataValue,
                        CacheRecord.TIME_NOT_AVAILABLE, origin, completionId));
            }
        } finally {
            onDeleteRecord(key, removedRecord);
        }

        return removedRecord != null;
    }

```